SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/errors/CompilerMessages.java

```diff
@@ -0,0 +1,196 @@
+package org.dbsp.sqlCompiler.compiler.errors;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.apache.calcite.runtime.CalciteContextException;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.dbsp.sqlCompiler.compiler.CompilerOptions;
+import org.dbsp.util.Unimplemented;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class CompilerMessages {
+    public class Error {
+        public final SourcePositionRange range;
+        public final boolean warning;
+        public final String errorType;
+        public final String message;
+
+        Error(SourcePositionRange range, boolean warning, String errorType, String message) {
+            this.range = range;
+            this.warning = warning;
+            this.errorType = errorType;
+            this.message = message;
+        }
+
+        Error(SqlParseException e) {
+            this.range = new SourcePositionRange(e.getPos());
+            this.warning = false;
+            this.errorType = "Error parsing SQL";
+            this.message = e.getMessage();
+        }
+
+        Error(CalciteContextException e) {
+            this.range = new SourcePositionRange(
+                    new SourcePosition(e.getPosLine(), e.getPosColumn()),
+                    new SourcePosition(e.getEndPosLine(), e.getEndPosColumn()));
+            this.warning = false;
+            this.errorType = "Error in SQL statement";
+            String message;
+            if (e.getCause() != null)
+                message = e.getCause().getMessage();
+            else if (e.getMessage() != null)
+                message = e.getMessage();
+            else
+                message = "";
+            this.message = Objects.requireNonNull(message);
+        }
+
+        Error(Throwable e) {
+            this.range = SourcePositionRange.invalid;
+            this.warning = false;
+            this.errorType = "This is a bug in the compiler (please report it to the developers)";
+            this.message = e.getMessage();
+        }
+
+        Error(Unimplemented e) {
+            if (e.object != null) {
+                if (e.object instanceof SqlNode) {
+                    SqlNode node = (SqlNode)e.object;
+                    this.range = new SourcePositionRange(node.getParserPosition());
+                } else {
```

Comment:
Does this case check exist because there are things further down in the IR that are unimplemented that don't have anything to do with handling the Calcite AST?

Reply:
Calcite has multiple representations, SqlNode is one, but RelNode is another.
Unfortunately many nodes that appear in the Rel representation actually do not have source code information in them.
I think that's a bug... 
For example RexLiteral.
In general, we'll add here cases for whatever Calcite IR classes that have source code information.